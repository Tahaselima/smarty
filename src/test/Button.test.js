import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

it('button component click event is correct work', (done) => {
  function handleClick() {
    done();
  }

  const { getByTestId } = render(
    <button data-testid="test-button" onClick={handleClick}>TÄ±kla!</button>
  );

  const node = getByTestId("test-button");

  fireEvent.click(node);
})

test('renders button component with props', () => {
  const { asFragment } = render(<button id="textButton" className="btn">Taha Selim!</button>);
  expect(asFragment()).toMatchSnapshot()
});

afterEach(cleanup);