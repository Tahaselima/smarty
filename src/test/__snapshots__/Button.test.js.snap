// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button component with props 1`] = `
<DocumentFragment>
  <button
    class="btn"
    id="textButton"
  >
    Taha Selim!
  </button>
</DocumentFragment>
`;
